### PR TITLE
Add tests for changelog generation

### PR DESCRIPTION
**Pull request by 6543**
and add editorconfig config + linting

### CODE REVIEW

File: src/utils/change.test.ts [hunk 1]
```diff
@@ -86,103 +86,130 @@ describe('change', () => {
...
+      const changelog = updateChangelogSection(latestVersion, nextVersion, oldChangelog, newSection);
+
+      expect(changelog).toMatchSnapshot();
+    });
+  });
 
-    tests.forEach(({ commitMessage, expectedVersion }) => {
-      expect(extractVersionFromCommitMessage(commitMessage)).toBe(expectedVersion);
+  describe('version extraction', () => {
+    it('should extract version from various commit messages', () => {
```
Comment (anbraten) [inline src/utils/change.test.ts:194]:
```suggestion
    it('should extract version from commit messages', () => {
```

File: src/utils/change.test.ts [hunk 2]
```diff
@@ -86,103 +86,130 @@ describe('change', () => {
...
-    },
-    {
-      name: 'should update existing changelog section',
-      file: '__fixtures__/CHANGELOG_2.md',
-      latestVersion: '2.0.1',
-      nextVersion: '2.0.3',
-    },
-    {
-      name: 'should remove versions newer than the latest released version',
-      // this happens if we bump the version in an existing release PR
```
Comment (anbraten) [inline src/utils/change.test.ts:146]: Why shouldn't we keep this note?

File: src/utils/change.test.ts [hunk 3]
```diff
@@ -86,103 +86,130 @@ describe('change', () => {
...
 
-    expect(nextVersion).toBe('2.0.0-rc.0');
-  });
+    describe('changelog text management', () => {
+      const forge = new GithubForge('', '');
+      const newSection = getChangeLogSection('1.0.0', 'v1.0.0', config, changes, forge, true);
 
-  it('should get the correct bump from a rc', () => {
-    const nextVersion = getNextVersionFromLabels('1.0.0-rc.2', config.user, changesWithPatch, false);
+      it('should handle empty/no changelog', () => {
```
Comment (anbraten) [inline src/utils/change.test.ts:132]:
```suggestion
      it('should add a new section to an empty changelog', () => {
```

File: src/utils/change.test.ts [hunk 4]
```diff
@@ -86,103 +86,130 @@ describe('change', () => {
...
+        const oldChangelog = '';
+        const changelog = updateChangelogSection('0.0.0', '1.0.0', oldChangelog, newSection);
+        expect(changelog).toBe(`# Changelog\n\n${newSection}\n`);
+      });
 
-    expect(nextVersion).toBe('1.0.0');
-  });
+      it('should handle changelog without pretext', () => {
```
Comment (anbraten) [inline src/utils/change.test.ts:138]:
```suggestion
      it('should update an existing changelog section', () => {
```

File: src/utils/change.test.ts [hunk 5]
```diff
@@ -86,103 +86,130 @@ describe('change', () => {
...
+  describe('changelog generation', () => {
+    it('should generate a changelog section', () => {
+      const forge = new GithubForge('', '');
+      const changelog = getChangeLogSection('1.0.0', 'v1.0.0', config, changes, forge, true);
+      expect(changelog).toMatchSnapshot();
+    });
 
-    expect(nextVersion).toBe('2.0.0-rc.0');
-  });
+    describe('changelog text management', () => {
```
Comment (anbraten) [inline src/utils/change.test.ts:128]: Don't need to go to crazy with nesting 😉 

```suggestion
```

File: .editorconfig [hunk 6]
```diff
@@ -0,0 +1,20 @@
+root = true
```
Comment (anbraten) [inline .editorconfig:1]: Aren't you using prettier? That should actually cover this already.

Comment (6543) [reply .editorconfig:1]: yes but IDEs just get it wrong without this config

Comment (6543) [reply .editorconfig:1]: (before you manually run prettier)

Comment (6543) [reply .editorconfig:1]: If you'd like i move that into an extra pull

**Review by anbraten — APPROVED** [review]

**Pull request merged**